src/hotspot/share/gc/parallel/psCardTable.cpp

```diff
@@ -122,13 +122,38 @@ class PSStripeShadowCardTable {
   const uint _card_shift;
   const uint _card_size;
   CardValue _table[PSCardTable::num_cards_in_stripe];
-  const CardValue* _table_base;
+  uintptr_t _table_base;
+
+  // Avoid UB pointer operations by using integers internally.
+
+  static_assert(sizeof(intptr_t) == sizeof(CardValue*), "simplifying assumption");
```

Comment:
Why check `sizeof(intptr_t)` instead of `uintptr_t` here?

Reply:
I mean, the change uses `uintptr_t` throughput, and probably the size of `intptr_t` and `uintptr_t` must be(?) the same, but why not check the actually used type?